### PR TITLE
Add sequence field to *JobInputFile models

### PR DESCRIPTION
**Pull request by johnbradley**
Migration populates sequence with id so legacy files will sort correctly.
Changes to support job file sorting: https://github.com/Duke-GCB/bespin-ui/issues/108

### CODE REVIEW

File: data/models.py [hunk 1]
```diff
@@ -334,7 +334,8 @@ class DDSJobInputFile(models.Model):
     dds_user_credentials = models.ForeignKey(DDSUserCredential, on_delete=models.CASCADE)
     destination_path = models.CharField(max_length=255, blank=False, null=True)
     size = models.BigIntegerField(null=False, blank=False, default=0, help_text='Size of file in bytes')
-
+    sequence = models.IntegerField(null=True, blank=False,
```
Comment (dleehr) [inline data/models.py]: Should the sequence number be unique within a stage group?

File: data/models.py [hunk 2]
```diff
@@ -349,6 +350,8 @@ class URLJobInputFile(models.Model):
     url = models.URLField(null=True)
     destination_path = models.CharField(max_length=255, blank=False, null=True)
     size = models.BigIntegerField(null=False, blank=False, default=0, help_text='Size of file in bytes')
+    sequence = models.IntegerField(null=True, blank=False,
```
Comment (dleehr) [inline data/models.py]: Should the sequence number be unique within a stage group?

Comment (johnbradley) [reply data/models.py]: Added unique constraint.

**Comment by johnbradley** [comment]
@dleehr Can you take another look at this PR?

File: data/migrations/0039_add_input_file_sequences.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+def populate_sequence_fields(apps, schema_editor):
+    populate_sequence_for_model(apps.get_model('data', 'DDSJobInputFile'))
+    populate_sequence_for_model(apps.get_model('data', 'URLJobInputFile'))
+
+
+def populate_sequence_for_model(model):
+    # Fill in sequence_group and sequence with numbers so we can get a baseline of sorting.
+    for input_file in model.objects.all():
+        input_file.sequence_group = input_file.id
+        input_file.sequence = 0  # first item in this sequence group
```
Comment (dleehr) [inline data/migrations/0039_add_input_file_sequences.py]: For migrating existing data where there's only one group of files (e.g. sample pairs), wouldn't it be more accurate to use a default `sequence_group` but variable `sequence`?

We discussed future UI where we'd partition stage groups by their `sequence_group`. This default would create a bunch of groups of size 1

File: data/api.py [hunk 4]
```diff
@@ -182,7 +182,7 @@ class DDSJobInputFileViewSet(viewsets.ModelViewSet):
     serializer_class = DDSJobInputFileSerializer
 
     def get_queryset(self):
-        return DDSJobInputFile.objects.filter(stage_group__user=self.request.user)
+        return DDSJobInputFile.objects.filter(stage_group__user=self.request.user).order_by('sequence')
```
Comment (dleehr) [inline data/api.py]: Did you consider ordering by `sequence_group` too?

File: data/api.py [hunk 5]
```diff
@@ -208,7 +208,7 @@ class URLJobInputFileViewSet(viewsets.ModelViewSet):
     serializer_class = URLJobInputFileSerializer
 
     def get_queryset(self):
-        return URLJobInputFile.objects.filter(stage_group__user=self.request.user)
+        return URLJobInputFile.objects.filter(stage_group__user=self.request.user).order_by('sequence')
```
Comment (dleehr) [inline data/api.py]: Did you consider ordering by `sequence_group` too?

**Review by dleehr — APPROVED** [review]

**Pull request merged**